src/main/java/uk/sky/cirrus/locking/Lock.java

```diff
@@ -0,0 +1,85 @@
+package uk.sky.cirrus.locking;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.locking.exception.CannotAcquireLockException;
+import uk.sky.cirrus.locking.exception.CannotReleaseLockException;
+
+import java.time.Duration;
+
+/**
+ * Each instance attempts to acquire the lock.
+ */
+public class Lock {
```

Comment:
Do we want the locks stuff in our public API as well? It means we won't be able to change them easily once people are using cqlmigrate. I'd prefer to try and keep this stuff package private or at least reduce the api surface substantially (everything in locking package appears public).